### PR TITLE
Fixes #60: - Add data source to determine device information enhancement

### LINKED ISSUE #60: Feature Request - Add data source to determine device information
We would like to be able to retreive device information such as host name, os version, hardware model, and cluster state.

I have investigated this and it should be pretty straightforward to implement a data_source that exposes the data retreived from `<get-system-information/>` and marshalling it into a struct which can be used populate the data source.

```
<get-system-information/>
<rpc-reply xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" xmlns:junos="http://xml.juniper.net/junos/19.3R0/junos">
<system-information>
<hardware-model>srx340</hardware-model>
<os-name>junos-es</os-name>
<os-version>19.3R2-S2.2</os-version>
<serial-number>ABCD123456</serial-number>
<host-name>Test-Hostname</host-name>
<cluster-node/>
</system-information>
```

### PR DESCRIPTION
**Pull request by tagur87**
FEATURES

* Add `junos_system_information` data source for accessing system information i.e. - Serial Number, Hardware Type, etc (Fixes #60)

### CODE REVIEW

File: junos/data_source_system_information.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+			},
+			"os_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"serial_number": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"host_name": {
```
Comment (jeremmfr) [inline junos/data_source_system_information.go:31]: `host_name` is not used. Hostname is filled in ID, can you remove this argument ?

File: website/junos.erb [hunk 2]
```diff
@@ -12,6 +12,9 @@
           <li<%= sidebar_current("docs-junos-data-source-interface") %>>
             <a href="/docs/providers/junos/d/interface.html">junos_interface</a>
           </li>
+          <li<%= sidebar_current("docs-junos-data-source-system-information") %>>
+            <a href="/docs/providers/junos/d/system_information.html">system_information</a>
```
Comment (jeremmfr) [inline website/junos.erb:16]: Name of link need to be `junos_system_information`

**Review by jeremmfr — CHANGES_REQUESTED** [review]
I added 2 comments need to be resolved before merging.

File: website/junos.erb (hunk 2 above)
Comment (tagur87) [reply website/junos.erb]: I updated as you specified. Thanks for the catch!

File: junos/data_source_system_information.go (hunk 1 above)
Comment (tagur87) [reply junos/data_source_system_information.go]: Thanks for catching this, i totally missed it. Thanks for the input, I have removed the attribute.

**Review by jeremmfr — APPROVED** [review]
Everything looks good, thanks for your contribution @tagur87

**Pull request merged**